tests/common/usage_helper.hpp

```
@@ -71,6 +71,13 @@ class MockSlaveUsage {
     return results.resource_usage(iteration++);
   }
 
+  process::Future<ResourceUsage> usageIter(int _iteration) {
+    if (_iteration >= results.resource_usage_size())
+      return ResourceUsage();
```

Don't you want to return an error?